python/langsmith/client.py

```diff
@@ -3577,6 +3592,39 @@ def _prepate_multipart_data(
 
         return encoder, data
 
+    def update_examples_multipart(
```

Comment:
`ls_schemas.ExampleUpdateWithAttachments` and `ls_schemas.ExampleUploadWithAttachments` should not have a `dataset_id`. Instead `dataset_id` should be passed as a param to the methods itself. This prevents you from having to check to see if every update/upload has the same `dataset_id`